### PR TITLE
Counter

### PR DESCRIPTION
**Pull request by Nassir-Al**
create a add function
create a delete function
create a search function
create a input function
create counter 
create "done function"

Es ist sehr viel und mit einem couch abgeklärt.

### CODE REVIEW

File: src/components/todoitem.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const TodoItem = ({id, done, title, onToggleTodo, onDeleteTodo}) => {
+	return (
+		<div>
+			<div key={id} className="TodoItem">
+				<input checked={done} onChange={() => onToggleTodo(id)} type="checkbox" />
+				<p>{title}</p>
+				<button onClick={() => onDeleteTodo(id)}>Löschen</button>
+			</div>
+		</div>
```
Comment (mbosselmann) [inline src/components/todoitem.js]: Please think about which semantic elements could be suitable for your component. Maybe you need a `form` element? 
As your `TodoItem` can be part of a list you work with `ul` and `li`. Therefore, you would add a `ul` element in your `TodoList` component and the `div` element in row three of your `TodoItem` component would be a `li` element. A solution I would prefer. ;)
You also don't need to add a key attribute. To add the key prop your `TodoList` component is enough for React to work correctly.
And please check how you write accessible forms. :)
And you should add a type your delete button, as the default type is `submit`, but a delete button is not a submit button. If you enter `type="button"` it should be fine. 👍

File: src/components/useStore.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+//import {v4 as uuid} from 'uuid';
```
Comment (mbosselmann) [inline src/components/useStore.js:1]: Out commented code should be removed. :)

File: src/pages/index.js [hunk 3]
```diff
@@ -1,15 +1,95 @@
...
+	// Suche von Todos
 	return (
-		<Layout>
-			<Head>
-				<title key="title">My Project</title>
-				<meta key="description" name="description" content="This is my project" />
-			</Head>
-			<h1>Home</h1>
-		</Layout>
+		<div className="main">
```
Comment (mbosselmann) [inline src/pages/index.js]: If this is a `main` element you should use the `main`tag. We are also working with styled components and not with css files. You should remove your classes.

File: src/pages/index.js [hunk 4]
```diff
@@ -1,15 +1,95 @@
...
+			<div>
```
Comment (mbosselmann) [inline src/pages/index.js]: Your input element should be wrapped in a form element. [Please check this MDN article.](https://developer.mozilla.org/en-US/docs/Learn/Forms/How_to_structure_a_web_form)

File: src/pages/index.js [hunk 5]
```diff
@@ -1,15 +1,95 @@
...
+				<input
+					onChange={event => setSearchQuery(event.target.value)}
+					type="text"
+					placeholder="search"
+				></input>
+			</div>
+
+			<div>
```
Comment (mbosselmann) [inline src/pages/index.js]: Why do you need a `div` here?

File: src/pages/index.js [hunk 6]
```diff
@@ -1,15 +1,95 @@
...
+				<h1>Smart Organize App</h1>
+			</div>
+
+			<div>
+				<div>
+					<input
```
Comment (mbosselmann) [inline src/pages/index.js]: Same here for your form and input.

File: src/pages/index.js [hunk 7]
```diff
@@ -1,15 +1,95 @@
...
+						onKeyDown={handleAddtodo}
+						type="text"
+						placeholder="Hier Termine Hinzufügen"
+					></input>
```
Comment (mbosselmann) [inline src/pages/index.js]: Your input element does not need a closing tag.

File: src/pages/index.js [hunk 8]
```diff
@@ -1,15 +1,95 @@
...
-			</Head>
-			<h1>Home</h1>
-		</Layout>
+		<div className="main">
+			<div>
+				<input
+					onChange={event => setSearchQuery(event.target.value)}
+					type="text"
+					placeholder="search"
+				></input>
```
Comment (mbosselmann) [inline src/pages/index.js]: Your input element does not need a closing tag.

File: src/pages/index.js [hunk 9]
```diff
@@ -1,15 +1,95 @@
...
+						/>
+						<TodoList
+							title="Erledigt"
+							list={doneTodos}
+							onDeleteTodo={handleDeleteTodo}
+							onToggleTodo={toggleTodo}
+						/>
+					</>
+				)}
+				<div>
```
Comment (mbosselmann) [inline src/pages/index.js]: Why do you need another `div` here?

**Review by mbosselmann — COMMENTED** [review]
Hi Nassir,
good job! 👍 Congratulations on your first pull request! 🎉 I left some remarks you should consider before you move further. :)

File: src/components/Counter.js [hunk 10]
```diff
@@ -0,0 +1,12 @@
+import {useState} from 'react';
+
+export default function Example() {
```
Comment (julialoeschel) [inline src/components/Counter.js:3]: yout function name is still Example - i guess this is not the name it should have ;)

File: src/components/todoitem.js [hunk 11]
```diff
@@ -0,0 +1,16 @@
+const TodoItem = ({id, done, title, onToggleTodo, onDeleteTodo}) => {
+	return (
+		<div>
+			<form className="TodoItem">
+				<input checked={done} onChange={() => onToggleTodo(id)} type="checkbox" />
+				<ul>
+					<li>{title}</li>
+				</ul>
```
Comment (julialoeschel) [inline src/components/todoitem.js:8]: do you really need a list here, if you only want to show the title?

File: src/pages/index.js [hunk 12]
```diff
@@ -1,15 +1,91 @@
...
 	return (
-		<Layout>
-			<Head>
-				<title key="title">My Project</title>
-				<meta key="description" name="description" content="This is my project" />
-			</Head>
-			<h1>Home</h1>
-		</Layout>
+		<>
+			<form>
```
Comment (julialoeschel) [inline src/pages/index.js:46]: please use a label for the input for accessability reasons ;)

**Review by julialoeschel — COMMENTED** [review]
good job Nassir! 

This could have been at least 4 PR (you know that 🐜 ).
please try to remember that next time.

**Pull request merged**